pkgs/by-name/me/meson/package.nix

```diff
@@ -90,6 +90,14 @@ python3.pkgs.buildPythonApplication rec {
     # Fix extraframework lookup on case-sensitive APFS.
     # https://github.com/mesonbuild/meson/pull/13038
     ./007-case-sensitive-fs.patch
+
+    # Fix meson's detection for zig's linker
+    # https://github.com/mesonbuild/meson/pull/12293
+    (fetchpatch {
+      name = "linker-support-zig-cc.patch";
+      url = "https://github.com/mesonbuild/meson/pull/12293/commits/2baae244c995794d9addfe6ed924dfa72f01be82.patch";
```

Comment:
Random commits in unmerged pull requests are not stable URLs.  If you must use a non-upstream patch like this, it has to be vendored to ensure it can still be built in future.

Reply:
This also doesn't look like it's going to be upstream any time soon.  I think this might have been a bit premature…

Reply:
Yeah, will pull this patch into #331011